agent/agenttest/client.go

```diff
@@ -254,3 +246,73 @@ type closeFunc func() error
 func (c closeFunc) Close() error {
 	return c()
 }
+
+type FakeAgentAPI struct {
```

Comment:
Could you generate this?

Reply:
I could generate it as a Mock, but that won't fit as easily into our existing tests.